js/apps/account-ui/src/personal-info/LocaleSelector.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import type { Option } from "ui-shared";
+import { SelectControl } from "ui-shared";
+import { supportedLocales } from "../api/methods";
+import { usePromise } from "../utils/usePromise";
+
+const localeToDisplayName = (locale: string) => {
+  try {
+    return new Intl.DisplayNames([locale], { type: "language" }).of(locale);
+  } catch (error) {
+    return locale;
+  }
+};
+
+export const LocaleSelector = () => {
+  const { t } = useTranslation();
+  const [locales, setLocales] = useState<Option[]>([]);
+
+  usePromise(
+    (signal) => supportedLocales({ signal }),
+    (locales) =>
+      setLocales(
+        locales.map(
+          (l) =>
+            ({
+              key: l,
+              value: localeToDisplayName(l),
+            } as Option)
+        )
```

Comment:
Instead of casting this value to the `Option` type, we should be able to provide a generic to the `map()` function instead.
```suggestion
        locales.map<Option>(
          (l) =>
            ({
              key: l,
              value: localeToDisplayName(l),
            })
        )
```